app/controllers/bag_controller.rb

```diff
@@ -1,12 +1,15 @@
 # frozen_string_literal: true
 
 class BagController < ApplicationController
+  skip_after_action :discard_flash_if_xhr
+
   def download
     send_file("#{Rails.application.config.bag_path}/#{bag_params}.zip", type: 'application/zip', disposition: 'attachment')
   end
 
   def create
-    BagJob.perform_later(work_ids: create_params)
+    BagJob.perform_later(work_ids: create_params, user: User.find(current_user.id))
```

Comment:
You don't need to look up the user here.  You should be able to just do this:
```
 BagJob.perform_later(work_ids: create_params, user: current_user)
```

Reply:
:+1: